src/S3/S3Transfer/S3TransferManager.php

```diff
@@ -1045,4 +1053,268 @@ public static function getDefaultConfig(): array
     {
         return self::$defaultConfig;
     }
+
+    /**
+     * @param array $source The object to copy, specified as an array with a 'Bucket' and 'Key' keys.
+     *                          Provide a 'VersionID' key to copy a specified version of an object
+     * @param array $copyRequestArgs Destination parameters including Bucket and Key
+     * @param array $config Additional configuration options
+     * @param array $listeners Array of transfer listeners
+     * @param ?TransferListener $progressTracker Optional progress tracker
+     * @return PromiseInterface
+     */
+    public function copy(
+        array $source,
+        array $copyRequestArgs,
+        array $config = [],
+        array $listeners = [],
+        ?TransferListener $progressTracker = null,
+    ): PromiseInterface
+    {
+        // Valid required parameters for both source and destination
+        $required = ['Bucket', 'Key'];
+
+        $this->validateRequiredParams($required, $source, 'source array');
+        $this->validateRequiredParams($required, $copyRequestArgs, 'copy request arguments');
+
+        // Validate buckets exist and are not identical objects
+        $this->validateNotSameObject(source: $source, dest: $copyRequestArgs);
+
+        $mupThreshold = $config['multipart_copy_threshold_bytes']
+            ?? $this->config['multipart_copy_threshold_bytes'];
+        if ($mupThreshold < AbstractMultipartUploader::PART_MIN_SIZE)
+        {
+            throw new InvalidArgumentException(
+                message: "The provided config `multipart_copy_threshold_bytes` "
+                . "must be greater than or equal to " . AbstractMultipartUploader::PART_MIN_SIZE
+            );
+        }
+        if ($progressTracker === null
+            && ($config['track_progress'] ?? $this->config['track_progress']))
+        {
+            $progressTracker = new SingleProgressTracker();
+        }
+
+        if ($progressTracker !== null)
+        {
```

Comment:
Generally if we're evaluating a single condition and it is within the maximum recommended line length, we will keep the opening brace on the same line as the closing parentheses.  We only do this by default in function bodies